vision/text/detect_words.py

```diff
@@ -15,68 +16,249 @@
+        #                                                  V
+        #                           Pass into pyTesseract, does it have our russian words in it?
+        #                                    Yes                                    No
+        #                                     |                                      |
+        #                                     V                                      V
+        #                              More trig to return the                   Were there anymore contours with children?
+        #                          bounding box relative to the whole image          Yes                         No
+        #                                    |                                       |                           |
+        #                                    V                                       V                           V
+        #                              That's it, you done         Go back through process with them         Either the image has no
+        #                                                                                                    russian or it is too noisy
+        #
+        # This would probably be wildly unperformant, but I think it would make pytesseract understand the rotated text
+
+        # the text is always in a blue rectangle. this finds the rectangle
+        # remove noise, int16 to prevent negative overflow in following step
+        blur_image = np.int16(cv2.GaussianBlur(color_image, (5, 5), 0))
+        b_image, g_image, r_image = (
+            blur_image[:, :, 0],
+            blur_image[:, :, 1],
+            blur_image[:, :, 2],
+        )  # separate channels
+
+        isblue = np.logical_and((b_image - r_image > 120), (b_image - g_image > 20))
+
+        # make blue parts black, rest white
+        mask_image = np.where(isblue, np.uint8(0), np.uint8(255))
+
+        # if we have depth data, get rid of irrelevant far aways stuff like the sky
+        if depth_image is not None:
+            mask_image = np.where((depth_image < 8000), mask_image, np.uint8(255))
+
+        edges = cv2.Canny(mask_image, 250, 255)
+        # canny is too good at its job, soften it up a bit for houghlines and findcontours
+        edges = cv2.GaussianBlur(edges, (5, 5), 0)
+
+        # specifically getting a 2-tier contour
+        # our target is the blue rectangle surrounding the text
+        # the rectangle has an inside and outside boundary, and we can use that fact to eliminate noise
+        # that is, any contours that don't have child contours aren't our rectangle
+        contours, hierarchy = cv2.findContours(
+            edges, cv2.RETR_CCOMP, cv2.CHAIN_APPROX_NONE
+        )
+
+        if hierarchy is None:
+            return None, None
+
+        largestArea = 0
+        largestContour = 0
+        for c in contours:
+            if cv2.contourArea(c) > largestArea:
+                largestContour = c
```

Comment:
This for loop is going to slow things down. Alternative is:

`contourAreas = contours[cv2.contourArea(c) for c in countours]`
`largestContour = contours[np.argmax(contourAreas)]`

Also it doesn't look like you were updating largestArea anyway? So you weren't getting the maximum.